kyaml/filesys/filesystem.go

```diff
@@ -16,40 +19,72 @@ const (
 // FileSystem groups basic os filesystem methods.
 // It's supposed be functional subset of https://golang.org/pkg/os
 type FileSystem interface {
+
 	// Create a file.
 	Create(path string) (File, error)
+
 	// MkDir makes a directory.
 	Mkdir(path string) error
+
 	// MkDirAll makes a directory path, creating intervening directories.
 	MkdirAll(path string) error
+
 	// RemoveAll removes path and any children it contains.
 	RemoveAll(path string) error
+
 	// Open opens the named file for reading.
 	Open(path string) (File, error)
+
 	// IsDir returns true if the path is a directory.
 	IsDir(path string) bool
+
 	// ReadDir returns a list of files and directories within a directory.
 	ReadDir(path string) ([]string, error)
+
 	// CleanedAbs converts the given path into a
 	// directory and a file name, where the directory
 	// is represented as a ConfirmedDir and all that implies.
 	// If the entire path is a directory, the file component
 	// is an empty string.
 	CleanedAbs(path string) (ConfirmedDir, string, error)
+
 	// Exists is true if the path exists in the file system.
 	Exists(path string) bool
+
 	// Glob returns the list of matching files,
 	// emulating https://golang.org/pkg/path/filepath/#Glob
 	Glob(pattern string) ([]string, error)
+
 	// ReadFile returns the contents of the file at the given path.
 	ReadFile(path string) ([]byte, error)
+
 	// WriteFile writes the data to a file at the given path,
 	// overwriting anything that's already there.
 	WriteFile(path string, data []byte) error
+
 	// Walk walks the file system with the given WalkFunc.
 	Walk(path string, walkFn filepath.WalkFunc) error
 }
 
+// DemandDir returns an error if the user-specified path is not an existing directory on fSys.
+// Otherwise, DemandDir returns path, which can be relative, as a ConfirmedDir and all that implies.
+func DemandDir(fSys FileSystem, path string) (ConfirmedDir, error) {
+	if path == "" {
+		return "", errors.Errorf("directory path cannot be empty")
+	}
+
+	d, f, err := fSys.CleanedAbs(path)
+	if err != nil {
+		return "", errors.WrapPrefixf(err, ErrNotDir.Error())
+	}
+	if f != "" {
+		return "", errors.WrapPrefixf(
+			errors.WrapPrefixf(errors.Errorf("file is not directory"), fmt.Sprintf("'%s'", path)),
```

Comment:
```suggestion
			errors.WrapPrefixf(errors.Errorf("%s is a file, not a directory", path)),
```

Reply:
`lint` will let me make this change because it doesn't have the `kyaml/errors` package that we've written pre-programmed. However, if I replaced `errors` with `fmt`, `lint` would complain that we should never create a dynamic error. Instead, we should wrap a static error with a dynamic error.

Given this, (just asking about best practice here) would you still prefer to change it?

Reply:
Oh I see, I didn't realize you were trying to avoid the lint error. Feel free to leave it as is, it just seemed really weird to me to have a a Sprintf within an Errorf within a WrapPrefix within a WrapPrefix. I was thinking it would be nice to simplify it, but if it leads to a lint error then we can leave it as is.

For best practices, making the linter happy takes precedence, and within those constraints I prefer simplicity.

Reply:
Understood, thanks for the explanation! In that case, I removed the outer layer of the error message to simplify this code.